base-theme/assets/js/utils.ts

```diff
@@ -39,3 +39,11 @@ const setOrGetLocalStorageItemHelper = (
     return false
   }
 }
+
+/**
+ * @return {boolean} returns true is app is running on local server, else false.
+ */
+export const isLocalServerRunning = () =>
```

Comment:
This will be okay to cover most cases, but a potential edge case could exist if someone has decided to add a `hosts` entry for something like `127.0.0.1 ocw.local`. Something we could add here to make this more robust would be to use the Hugo variable [`site.IsServer`](https://gohugo.io/variables/site/). We don't template our JS files through Hugo, so we wouldn't be able to pull it directly into this function, but we could inline something on the base template like:

```
    <script type="text/javascript">
    function isHugoDevServer() {
      return {{ site.IsServer }}
    }
    </script>
```
We could then add an extra line to the or statements here like:
```
  location.hostname === "localhost" ||
  location.hostname === "127.0.0.1" ||
  location.hostname === "" ||
  isHugoDevServer()
```

This wouldn't cover sites built through the Minio based local dev setup in `ocw-studio` because that all happens through nginx. We could maybe do something to detect that in the future, but I think checking `site.IsServer` is enough of an easy win here that we should do it.